src/ui/component/toast/toast.tsx

```diff
@@ -1,51 +1,140 @@
 import classNames from 'classnames'
-import { useEffect } from 'react'
+import type { FC } from 'react'
+import { useMemo, useCallback, useRef, useEffect, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import type { NotificationID, NotificationType } from '@/domain/notification/entity'
+import type { IconName } from '@/ui/component/icon/icon'
+import { Icon } from '@/ui/component/icon/icon'
+import { useOnClickOutside } from '@/ui/hook/useOnClickOutside'
+import { useOnKeyboard } from '@/ui/hook/useOnKeyboard'
+import { Button } from '@/ui/component//button/button'
+import type { DismissNotificationInput } from '@/domain/notification/aggregate'
+import '@/ui/component/notifications/i18n'
 import './toast.scss'
 
-type ToastVariants = 'success' | 'error' | 'warning' | 'info'
+type ToastCTAProps = {
+  type: NotificationType
+  action: string
+}
+const ToastCTA: FC<ToastCTAProps> = ({ action, type }) => {
+  const { t } = useTranslation('notifications')
+
+  const handleClick = useCallback(() => {
+    window.location.reload()
+  }, [])
+  return (
+    <Button
+      className={classNames('okp4-dataverse-portal-toast-cta', type)}
+      label={t(`action.${action}`)}
+      onClick={handleClick}
+    />
+  )
+}
 
 type ToastProps = {
+  id: NotificationID
+  type: NotificationType
   title: string
-  autoHideDuration?: number
-  description?: string
-  icon?: JSX.Element
-  onClose?: () => void
-  open?: boolean
-  variant?: ToastVariants
+  onClose: (input: DismissNotificationInput) => void
+  message?: string
+  iconName?: IconName
+  action?: string
+  index: number
 }
 
-export const Toast = ({
+// eslint-disable-next-line max-lines-per-function
+export const Toast: FC<ToastProps> = ({
+  id,
+  type,
   title,
-  autoHideDuration,
-  description,
-  icon,
   onClose,
-  open = false,
-  variant = 'info'
-}: ToastProps): JSX.Element | null => {
+  message,
+  iconName,
+  action,
+  index
+}) => {
+  const [isHidding, setIsHidding] = useState(false)
+  const containerRef = useRef<HTMLDivElement | null>(null)
+  const timeoutRef = useRef<ReturnType<typeof setTimeout>>()
+  const autoHideTimeoutRef = useRef<ReturnType<typeof setTimeout>>()
+  const animationDurationRef = useRef<number>(500)
+  const stackedPosition = useRef<number>(index + 1)
+
+  const autoHideDuration = useMemo((): number => {
+    const messagesLength = title.length + (message?.length ?? 0)
+    const minDuration = 2000
+    const maxDuration = 7000
+    const durationPerChar = 50
+    const duration = messagesLength * durationPerChar
+    return Math.min(Math.max(duration, minDuration), maxDuration)
+  }, [title, message])
```

Comment:
Have all these values been approved by the product team?

Reply:
No, I removed it in [6a30ad9](https://github.com/okp4/dataverse-portal/pull/195/commits/6a30ad9d5a347ed926a5c3e4d02a13788388b5fb).